### PR TITLE
Fix setting admission plugins on local-up-cluster.sh

### PR DESCRIPTION
**Pull request by jfchevrette**
**What this PR does / why we need it**:

`hack/local-up-cluster.sh` currently do not honor setting `ENABLE_ADMISSION_PLUGINS` because the list is overriden later on in the script before the apiserver is started.

This PR moves the list higher up as a default list of admission plugins and then allows overriding the defaults via the expected ENV variable when starting the cluster.

**Which issue(s) this PR fixes**
No issue created. Should I create one?

**Special notes for your reviewer**:
I have not found tests for this script. I did run it and was able to manually validate that the list of admission plugin is either set to default or to those I chose via the environment variable.

This is a first contribution. I will appreciate any comments on how I can improve future contributions.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by jfchevrette** [comment]
/assign @dims
PTAL

File: hack/local-up-cluster.sh [hunk 1]
```diff
@@ -103,7 +103,11 @@ export KUBE_CACHE_MUTATION_DETECTOR
 KUBE_PANIC_WATCH_DECODE_ERROR="${KUBE_PANIC_WATCH_DECODE_ERROR:-true}"
 export KUBE_PANIC_WATCH_DECODE_ERROR
 
-ENABLE_ADMISSION_PLUGINS=${ENABLE_ADMISSION_PLUGINS:-""}
+# Default list of admission Controllers to invoke prior to persisting objects in cluster
+# The order defined here does not matter.
+DEFAULT_ENABLE_ADMISSION_PLUGINS="LimitRanger,ServiceAccount,DefaultStorageClass,DefaultTolerationSeconds,MutatingAdmissionWebhook,ValidatingAdmissionWebhook,ResourceQuota,StorageObjectInUseProtection"
```
Comment (yue9944882) [inline hack/local-up-cluster.sh]: the default on plugins are defined in:[https://github.com/kubernetes/kubernetes/blob/72dd54ec4856ccb7bd2b08258d95f3d0b932fb11/pkg/kubeapiserver/options/plugins.go#L132-L142](https://github.com/kubernetes/kubernetes/blob/72dd54ec4856ccb7bd2b08258d95f3d0b932fb11/pkg/kubeapiserver/options/plugins.go#L132-L142)

so we can trim the list, amiright?

File: hack/local-up-cluster.sh [hunk 2]
```diff
@@ -507,11 +511,8 @@ function start_apiserver {
       RUNTIME_CONFIG+="scheduling.k8s.io/v1alpha1=true"
     fi
 
-
-    # Admission Controllers to invoke prior to persisting objects in cluster
-    #
-    # The order defined here dose not matter.
-    ENABLE_ADMISSION_PLUGINS=LimitRanger,ServiceAccount${security_admission},DefaultStorageClass,DefaultTolerationSeconds,MutatingAdmissionWebhook,ValidatingAdmissionWebhook,ResourceQuota,StorageObjectInUseProtection
```
Comment (yue9944882) [inline hack/local-up-cluster.sh]: what about removing L106 and change this line to
```
ENABLE_ADMISSION_PLUGINS=${ENABLE_ADMISSION_PLUGINS:-LimitRanger,ServiceAccount${security_admission},DefaultStorageClass,DefaultTolerationSeconds,MutatingAdmissionWebhook,ValidatingAdmissionWebhook,ResourceQuota,StorageObjectInUseProtection}
```

**Review by yue9944882 — COMMENTED** [review]
thx so much for fixing this @jfchevrette ! if memory serves, i hit the exact same problem before. 🎉

File: hack/local-up-cluster.sh (hunk 1 above)
Comment (jfchevrette) [reply hack/local-up-cluster.sh]: The only plugin that is in the list but not in the default on plugins list is `StorageObjectInUseProtection` which according to the [cli feature gates docs](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/) is enabled by default. 

Regardless of the above, as a user using local-up-cluster.sh I would expect that the default plugins are loaded unless I specify additional plugins using ENABLE_ADMISSION_PLUGINS (or disable plugins using DISABLE_ADMISSION_PLUGINS)

WDYT?

File: hack/local-up-cluster.sh (hunk 2 above)
Comment (jfchevrette) [reply hack/local-up-cluster.sh]: I've removed the additional `DEFAULT_` variable but left (and updated) the list to match the recommended [defaults from the docs](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#is-there-a-recommended-set-of-admission-controllers-to-use)

It is my understanding that if --enable-admission-plugins is passed with an empty value/list, no plugins will be loaded.

**Comment by jfchevrette** [comment]
@yue9944882 I just found out that kube-apiserver / hyperkube can return the list of default plugins. We can use that to generate the list dynamically and that would avoid the need to keep a static list up-to-date. 

```
hyperkube apiserver -h | grep -E '^\s+\-\-enable\-admission\-plugins' | sed  's/.*(\(.*\)).*/\1/' | tr -d ' '
```

**Pull request merged**